microsoft/nni
# Refactor NNI manager globals (step 4) - logging

### Description

Make logging system global and robust.

It also changed a behavior that when using `foreground=True`, logs will be written to stdout AND nnimanager.log.
Previously in foreground mode it does not write to log file.

### Checklist


### How to Test

Run ordinary experiments in foreground and background mode, then check log file.

### Release Note

- [Experiment] Foreground mode now saves `nnimanager.log`.
